sdk/python/kfp/dsl/tasks_group.py

```diff
@@ -140,7 +143,16 @@ def __init__(
         # Set is_exit_handler since the compiler might be using this attribute.
         exit_task.is_exit_handler = True
 
-        self.exit_task = exit_task
+    def __has_dependent_tasks(self) -> bool:
+        if self.exit_task.dependent_tasks:
+            return True
+
+        if not self.exit_task.inputs:
+            return False
+
+        return any(
+            type(task_input) is PipelineParameterChannel
```

Comment:
i believe it's more pythontic to use `isinstance` here instead of manually checking type

Reply:
It depends. `type` will assert without inheritance. If it is the case, then it is fine.

Reply:
> `type` will assert without inheritance.

That said, I think we need `isinstance` here.